dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java

```diff
@@ -78,6 +80,21 @@ public class JdbcIconStore implements IconStore {
 
   private final FileResourceStore fileResourceStore;
 
+  @Override
+  public List<String> getKeysWithoutIcon(Set<String> keys) {
+    String sql =
+        """
+    SELECT ARRAY(
+                   SELECT UNNEST(ARRAY [:keys])
+                   except
+                   SELECT icon.iconkey from icon)""";
```

Comment:
I am no SQL guru - this is the best I could find to return only those keys that do not have a row in icon